### PR TITLE
Deletion Language is in english

### LINKED ISSUE #4800: Deletion requests: Post in English rather than in the flagger's locale
An app user (using the Romanian locale on their phone) flagged a picture for deletion.
The deletion requests was posted in Romanian:

https://commons.wikimedia.org/w/index.php?title=File%3AHexagone-Balard_%40_Paris_%2823766157612%29.jpg&type=revision&diff=627919510&oldid=627919420

The picture's uploader probably does not understand Romanian.
Even though we can not expect everyone to understand English, the current norm on Commons is to post these notices in English.

We should thus un-internationalize this message.

The month should also not be localized (`februarie` in the example).

### PR DESCRIPTION
**Pull request by devarsh-mavani-19**
**Description (required)**
When uploading a deletion request all the text should be in English rather than user's locale

Fixes #4800 

**Tests performed (required)**
Device: Redmi 6 Pro
Build Beta
API level 28

### CODE REVIEW

**Comment by devarsh-mavani-19** [comment]
@madhurgupta10 can you please help me with powermock? I am getting classnotprepared exception even though I have added @PrepareForTest(AccountUtil::class) annotation. I am using power mock so that I can use AccountUtil.getUserName static method.

**Comment by madhurgupta10** [comment]
> @madhurgupta10 can you please help me with powermock? I am getting classnotprepared exception even though I have added @PrepareForTest(AccountUtil::class) annotation. I am using power mock so that I can use AccountUtil.getUserName static method.



> @madhurgupta10 can you please help me with powermock? I am getting classnotprepared exception even though I have added @PrepareForTest(AccountUtil::class) annotation. I am using power mock so that I can use AccountUtil.getUserName static method.

You don't need to use PowerMock, see `AccountUtilUnitTest` for ref

**Comment by devarsh-mavani-19** [comment]
@madhurgupta10 in the account util tests there are only assert statements. But I want the accountutils.getusername method to return a username for that currently I am using 
` `when`(AccountUtil.getUserName(context)).thenReturn("anonymous")`

but it is giving error since I think I cannot mock static method. I want the AccountUtil.getUserName(context) to return a name.

**Comment by madhurgupta10** [comment]
> @madhurgupta10 in the account util tests there are only assert statements. But I want the accountutils.getusername method to return a username for that currently I am using ` `when`(AccountUtil.getUserName(context)).thenReturn("anonymous")`
> 
> but it is giving error since I think I cannot mock static method. I want the AccountUtil.getUserName(context) to return a name.

You can create a wrapper, similar to `FakeContextWrapper` for the context and return whatever you need.

**Comment by devarsh-mavani-19** [comment]
> > @madhurgupta10 in the account util tests there are only assert statements. But I want the accountutils.getusername method to return a username for that currently I am using ` `when`(AccountUtil.getUserName(context)).thenReturn("anonymous")`
> > but it is giving error since I think I cannot mock static method. I want the AccountUtil.getUserName(context) to return a name.
> 
> You can create a wrapper, similar to `FakeContextWrapper` for the context and return whatever you need.

Somthing like this right?

        val mContext = FakeContextWrapper(RuntimeEnvironment.application.applicationContext)
        `when`(context).thenReturn(mContext)

**Comment by madhurgupta10** [comment]
Here the context is not a mock so you can't use `when-then`, wrapper simply inherits the properties of passed in context and add additional ones you provide.

Simply use
```
context = YourWrapper(RuntimeEnvironment.application.applicationContext)
```

**Comment by devarsh-mavani-19** [comment]
@madhurgupta10 I did as you said still the tests are failing.. In fact all the tests are now failing of MediaDetailFragmentUnitTests class. Can you please check the last commit I made? I am not sure what I did wrong

**Comment by madhurgupta10** [comment]
@devarsh-mavani-19 Please try to debug locally before pushing commits, you can use Android Studio to run your tests.

**Comment by devarsh-mavani-19** [comment]
@madhurgupta10 ok I will debug locally.

**Comment by devarsh-mavani-19** [comment]
> @devarsh-mavani-19 Please try to debug locally before pushing commits, you can use Android Studio to run your tests.

Can you help me with why the previous tests keep failing with MockitoException on         MockitoAnnotations.initMocks(this)
line?

**Comment by devarsh-mavani-19** [comment]
@madhurgupta10 I am able to get rid of most of problem. The only problem I am facing is that when I call fragment.onDeleteClick, it starts executing it and when it encoutners a method reasonBuilder.getReason() it is executing real method rather than returning mock result. the reasonBuilder is mock object. here is the test function. i tried looking on internet but couldn't found anything helpful. can you please check this? Thanks

val spinner = mock(Spinner::class.java)
        `when`(media.imageUrl).thenReturn("test@example.com")
        `when`(spinner.selectedItemPosition).thenReturn(0)
        `when`(reasonListEnglishMappings?.get(spinner.selectedItemPosition)).thenReturn("TESTING")
        `when`(applicationKvStore.getBoolean(String.format(MediaDetailFragment.NOMINATING_FOR_DELETION_MEDIA,media.imageUrl
                ))).thenReturn(true)
        doReturn(Single.just(true)).`when`(deleteHelper).makeDeletion(ArgumentMatchers.any(),ArgumentMatchers.any(), ArgumentMatchers.any())

        doReturn(Single.just("")).`when`(reasonBuilder).getReason(ArgumentMatchers.any(), ArgumentMatchers.any())

        fragment.onDeleteClicked(spinner)

**Comment by madhurgupta10** [comment]
You need to replace the `reasonBuilder` object with a mock using `setInternalState`

File: app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java [hunk 1]
```diff
@@ -1198,12 +1209,13 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
         }
 
     @SuppressLint("CheckResult")
-    private void onDeleteClicked(Spinner spinner) {
+    public void onDeleteClicked(Spinner spinner) {
```
Comment (madhurgupta10) [inline app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java:1212]: Any specific reason why this needs to be public?

File: app/build.gradle [hunk 2]
```diff
@@ -83,6 +83,7 @@ dependencies {
     testImplementation "org.powermock:powermock-module-junit4:2.0.2"
     testImplementation "org.powermock:powermock-api-mockito2:2.0.2"
 
+
```
Comment (madhurgupta10) [inline app/build.gradle:86]: Can you revert this change?

**Review by madhurgupta10 — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 2 above)
Comment (devarsh-mavani-19) [reply app/build.gradle:86]: ya sure

File: app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java (hunk 1 above)
Comment (devarsh-mavani-19) [reply app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java:1212]: actually wanted to do it locally but instead pushed it by mistake I will make it private

**Comment by devarsh-mavani-19** [comment]
@madhurgupta10 all tests are passing thanks a lot for helping

**Review by madhurgupta10 — APPROVED** [review]
Haven't tested but the changes look good to me. I will request @nicolas-raoul for the review :)

**Review by nicolas-raoul — APPROVED** [review]
Tested in Japanese locale with a selfie, the posted message is indeed in English, thanks! https://commons.wikimedia.org/wiki/File:Aqua_krishna.jpg
